pipeline/azure-pipelines.yml

```diff
@@ -0,0 +1,215 @@
+  - task: DownloadSecureFile@1
+    name: signKey
+    displayName: Download sign key
+    condition: and(succeeded(), eq(variables['Build.SourceBranchName'], 'master'))
+    inputs:
+      secureFile: 3755ac9b-4269-4218-a8d7-97aa4f1dd117
+  - task: PowerShell@2
+    displayName: Rename and sign SBOM file
+    condition: and(succeeded(), eq(variables['Build.SourceBranchName'], 'master'))
+    inputs:
+      targetType: filePath
+      filePath: $(System.DefaultWorkingDirectory)\pipeline\scripts\sbom.ps1
+  - task: CopyFiles@2
+    displayName: 'Copy VSIX to: $(Build.ArtifactStagingDirectory)'
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)\binaries
+      Contents: >-
+        **\*.vsix
+
+        **\SonarLint.visualstudio.sbom*.*
+      TargetFolder: $(Build.ArtifactStagingDirectory)
+  - task: CopyFiles@2
+    displayName: 'Copy any MSBuild binary logs to: $(Build.ArtifactStagingDirectory)'
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)
+      Contents: '**\*.binlog'
+      TargetFolder: $(Build.ArtifactStagingDirectory)
+  - task: CopyFiles@2
+    displayName: 'Copy marketplace files to:  $(Build.ArtifactStagingDirectory) (VS2022 only)'
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2022'))
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)\MarketPlaceFiles\VS2022
+      TargetFolder: $(Build.ArtifactStagingDirectory)
+  - task: CopyFiles@2
+    displayName: 'Copy marketplace files to:  $(Build.ArtifactStagingDirectory) (VS2019 only)'
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2019'))
+    inputs:
+      SourceFolder: $(Build.SourcesDirectory)\MarketPlaceFiles\VS2019
+      TargetFolder: $(Build.ArtifactStagingDirectory)
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact: vsix'
+    inputs:
+      ArtifactName: vsix
+  - task: VSTest@2
+    displayName: VsTest - testAssemblies
+    inputs:
+      testAssemblyVer2: >-
+        **\$(BuildConfiguration)\**\*test*.dll
+
+        !**\obj\**
+      runInParallel: false
+      runTestsInIsolation: false
+      codeCoverageEnabled: true
+      platform: $(BuildPlatform)
+      configuration: $(BuildConfiguration)
+      rerunFailedTests: false
+  - task: SonarCloudAnalyze@1
+    displayName: Run Code Analysis (VS2022 only)
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2022'))
+  - task: SonarCloudPublish@1
+    displayName: Publish Quality Gate Result (VS2022 only)
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2022'))
+  - task: PowerShell@2
+    displayName: Set trigger dogfood VSIX build tag (VS2022 only)
+    condition: and(succeeded(), eq(variables['vsTargetVersion'], '2022'))
+    inputs:
+      targetType: filePath
+      filePath: $(System.DefaultWorkingDirectory)\pipeline\scripts\trigger-dogfood.ps1
+      arguments: -SourceBranch $(Build.SourceBranch) -TriggerDogfoodReleaseIfMasterBranch $(TriggerDogfoodReleaseIfMasterBranch)
```

Comment:
variables are not accessible in files. So they need to be passed as parameters.

Reply:
I thought variables were made available to PowerShell scripts as environment variables (and some of the scripts refer to pre-defined variables e.g. `$env:SYSTEM_TEAMPROJECTID`).

However, there's no harm in being explicit about which variables are being passed.

Reply:
This was the solution I was able to find. I think using environment variables is cleaner, so I'll move to that.